### PR TITLE
tsduck: init at 3.19-1520

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change
Wanted to test video stream
@dasJ 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/video/tsduck/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{ stdenv, fetchFromGitHub, fetchurl, makeWrapper
+, dtapi, curl, linuxHeaders }:
+
+stdenv.mkDerivation rec {
+  pname = "tsduck";
+  version = "3.18-1361";
```
Comment (ajs124) [inline pkgs/tools/video/tsduck/default.nix]: The most current upstream release seems to be  version 3.19-1520, is there any reason why we should init this at an out-of-date version?

**Review by dasJ — APPROVED** [review]

**Comment by Lassulus** [comment]
```
hash mismatch in fixed-output derivation '/nix/store/hr4723z3p1n0rkcqxwvkc9rvfl7g79ji-source':
  wanted: sha256:1k80dy16b5093qd6lx977b5v8jgdmb0h25jvlgi2gpl6l77c0s0b
  got:    sha256:1w7qq4hjk144dwqhfrkbf7qgz1i8pf8z8phyk9gwc4yd9i5rx8an
cannot build derivation '/nix/store/cvhq2js30vaksfrh5y2x56rf17ml3w4y-tsduck-3.19-1520.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/yp6xlwz3igma8dyzmbb2kcvsw0b3ad8r-env.drv': 1 dependencies couldn't be built
```

**Review by dasJ — CHANGES_REQUESTED** [review]

**Comment by Lassulus** [comment]
```
builder for '/nix/store/1p2fja8d3i583sfkn68gznmlhmvsi9si-tsduck-3.19-1520.drv' failed with exit code 2; last 10 log lines:
  configuring
  no configure script, doing nothing
  building
  build flags: SHELL=/nix/store/715fljdrspphbdk255r729zxp4w7lcgk-bash-4.4-p23/bin/bash NOPCSC=1 LDFLAGS_EXTRA=/nix/store/wwaln2rihi440sybf79ij0lif73sm8ic-dtapi-2019.11.0/Lib/GCC5.1_CXX11_ABI1/DTAPI64.o
  ./dtapi-config.sh --url >LinuxSDK.version
  curl: (6) Could not resolve host: www.dektec.com
  curl: (6) Could not resolve host: www.dektec.com
  dtapi-config.sh: cannot locate LinuxSDK location from Dektec
  make[1]: *** [Makefile:71: geturl] Error 1
  make: *** [Makefile:58: default] Error 2
cannot build derivation '/nix/store/i3g93hp5y6j5dsyxr5h646y6msiilibb-env.drv': 1 dependencies couldn't be built
```

**Comment by Scriptkiddi** [comment]
@Lassulus

**Review by dasJ — CHANGES_REQUESTED** [review]
Please refrain from adding me as maintainer of random packages.

**Comment by Scriptkiddi** [comment]
> Please refrain from adding me as maintainer of random packages.

done

**Comment by Lassulus** [comment]
```
builder for '/nix/store/5cb1xhnz1w5jji5jbw3jaw7asnlk7r3f-tsduck-3.19-1520.drv' failed with exit code 2; last 10 log lines:
  configuring
  no configure script, doing nothing
  building
  build flags: SHELL=/nix/store/2jysm3dfsgby5sw5jgj43qjrb5v79ms9-bash-4.4-p23/bin/bash NOPCSC=1 LDFLAGS_EXTRA=/nix/store/h9ws6npv22znvnw73xyxfya0mh0jkqsa-dtapi-2020.08.0/Lib/GCC5.1_CXX11_ABI1/DTAPI64.o
  ./dtapi-config.sh --url >LinuxSDK.version
  curl: (6) Could not resolve host: www.dektec.com
  curl: (6) Could not resolve host: www.dektec.com
  dtapi-config.sh: cannot locate LinuxSDK location from Dektec
  make[1]: *** [Makefile:71: geturl] Error 1
  make: *** [Makefile:58: default] Error 2
cannot build derivation '/nix/store/yqqqvm54pp7lhvgp1fqib799ycx2pg6c-env.drv': 1 dependencies couldn't be built
[0 built (1 failed), 0.0 MiB DL]
```

**Comment by dasJ** [comment]
Also, 3.23-1965 is out now

File: pkgs/development/libraries/dtapi/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  postPatch = ''
+    substituteInPlace DTAPI/Include/DTAPI.h --replace /usr/include/ ""
+  '';
+
+  installPhase = ''
+    mkdir $out
+    cp License $out
+    cd DTAPI
+    cp -r Include/ Lib/ $out
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/dtapi/default.nix:20]:
```suggestion
    cp -r DTAPI/Include/ DTAPI/Lib/ $out
```

File: pkgs/tools/video/tsduck/default.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    rev = "v${version}";
+    sha256 = "1w7qq4hjk144dwqhfrkbf7qgz1i8pf8z8phyk9gwc4yd9i5rx8an";
+  };
+
+  postPatch = ''
+    patchShebangs .
+
+    # Provide patched dtapi
+    mkdir -p dektec/LinuxSDK/DTAPI
+    cp -r ${dtapi}/* dektec/LinuxSDK/DTAPI/
```
Comment (SuperSandro2000) [inline pkgs/tools/video/tsduck/default.nix:20]: Please use symlink to reduce the final size.

File: pkgs/tools/video/tsduck/default.nix [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    # Disable automatic downloading or unpacking of dtapi
+    sed -i 's/^dtapi:.*/dtapi:/g' dektec/Makefile
+  '';
+
+  buildInputs = [ curl linuxHeaders ];
+  nativeBuildInputs = [ curl makeWrapper ]; # Makefile calls curl-config
```
Comment (SuperSandro2000) [inline pkgs/tools/video/tsduck/default.nix:27]:
```suggestion
  # Makefile calls curl-config
  nativeBuildInputs = [ curl makeWrapper ];

  buildInputs = [ curl linuxHeaders ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
```
builder for '/nix/store/xpyzhkmc09zrghm6zl12w8hp8mdppcmm-tsduck-3.19-1520.drv' failed with exit code 2; last 10 log lines:
  configuring
  no configure script, doing nothing
  building
  build flags: SHELL=/nix/store/516z50fm1jbpcl32qnzy7kynrh0vl22w-bash-4.4-p23/bin/bash NOPCSC=1 LDFLAGS_EXTRA=/nix/store/jdlxw7fpifh0zwp2ad98cy95fs7gxgm0-dtapi-2020.08.0/Lib/GCC5.1_CXX11_ABI1/DTAPI64.o
  ./dtapi-config.sh --url >LinuxSDK.version
  curl: (6) Could not resolve host: www.dektec.com
  curl: (6) Could not resolve host: www.dektec.com
  dtapi-config.sh: cannot locate LinuxSDK location from Dektec
  make[1]: *** [Makefile:71: geturl] Error 1
  make: *** [Makefile:58: default] Error 2
cannot build derivation '/nix/store/vh5sv9pd1wc8qywmagdy43515s9zkazx-env.drv': 1 dependencies couldn't be built
```

**Comment by Lassulus** [comment]
this PR is comedy

**Pull request closed without merging**